Sources/ConcurrencyHelpers/Protected.swift

```diff
@@ -0,0 +1,36 @@
+/// Thread safe access to simple variable protected by lock
+@propertyWrapper
+public final class Protected<T> {
```

Comment:
"Synchronized" ?

For me "protected" is smth related to access control.

Reply:
It's kind of cleaner when do you use it imho:
```
@Protected
var continuation: AsyncStream<PermanentMessage>.Continuation? = nil
```
vs 
```
@Synchronized
var continuation: AsyncStream<PermanentMessage>.Continuation? = nil
```
It's not clear what ```Synchronized``` means in this context? Synchronized with what?

Another option:
```
@ThreadSafe
var continuation: AsyncStream<PermanentMessage>.Continuation? = nil
```

@hassila, @dimlio, @ser-0xff, what do you think?

Reply:
My inspiration is https://github.com/facebook/folly/blob/main/folly/docs/Synchronized.md
+ "synchronized" in Java.